src/common/io_exerciser/JsonStructures.h

```diff
@@ -0,0 +1,321 @@
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "include/types.h"
+
+#include "OpType.h"
+
+/* Overview
+ *
+ * class JSONStructure
+ *   Stores elements of a JSONStructure in C++ friendly format so they do not
+ *   have to be parsed from strings. Includes encode and decode functions to
+ *   provide easy ways to convert from c++ structures to json structures.
+ *
+ */
+
+class JSONObj;
+
+namespace ceph
+{
```

Comment:
Note that the formatting used here, and on many other constructs, is not the common Ceph one.
I'm not going to comment on this issue further (partially - because most can be fixed later, if so desired,
by running clang-format).